src/main/java/beam/analysis/plots/DeadHeadingStats.java

```diff
@@ -127,108 +214,73 @@ private boolean updateDeadHeadingTNCMap(double length,int hour,Integer _num_pass
-        boolean legend = true;
-        final JFreeChart chart = GraphUtils.createStackedBarChartWithDefaultSettings(dataset, graphTitle, deadHeadingTNC0XAxisTitle, deadHeadingTNC0YAxisTitle, fileName, legend);
-        CategoryPlot plot = chart.getCategoryPlot();
-        List<String> legendItemList= getLegendItemList(graphName,dataset.getRowCount(),getBucketSize());
-        GraphUtils.plotLegendItems(plot,legendItemList,dataset.getRowCount());
-        String graphImageFile = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(iterationNumber, fileName);
-        GraphUtils.saveJFreeChartAsPNG(chart, graphImageFile, GraphsStatsAgentSimEventsListener.GRAPH_WIDTH, GraphsStatsAgentSimEventsListener.GRAPH_HEIGHT);
+        writeToCSV(event.getIteration(), GraphsStatsAgentSimEventsListener.TNC_DEAD_HEADING_DISTANCE);
+
+
+        // Updating the model for the RideHailStats.csv
+        updateRideHailStatsModel(event);
+        writeRideHailStatsCSV(event);
     }
 
-    private void createDeadHeadingGraph(CategoryDataset dataset, int iterationNumber, String graphName) throws IOException {
-        String fileName = getFileName(graphName,"png");
-        String graphTitle = getTitle(graphName);
-        boolean legend = true;
-        final JFreeChart chart = GraphUtils.createStackedBarChartWithDefaultSettings(dataset, graphTitle, deadHeadingXAxisTitle, deadHeadingYAxisTitle, fileName, legend);
-        CategoryPlot plot = chart.getCategoryPlot();
-        List<String> legendItemList= getLegendItemList(graphName,dataset.getRowCount(),getBucketSize());
-        GraphUtils.plotLegendItems(plot,legendItemList,dataset.getRowCount());
-        String graphImageFile = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(iterationNumber, fileName);
-        GraphUtils.saveJFreeChartAsPNG(chart, graphImageFile, GraphsStatsAgentSimEventsListener.GRAPH_WIDTH, GraphsStatsAgentSimEventsListener.GRAPH_HEIGHT);
+    private void updateRideHailStatsModel(IterationEndsEvent event){
+        RideHailDistanceRowModel model = GraphUtils.RIDE_HAIL_REVENUE_MAP.getOrDefault(event.getIteration(), new RideHailDistanceRowModel());
+
+        model.setPassengerVkt(passengerVkt);
+        model.setDeadheadingVkt(deadHeadingVkt);
+        model.setRepositioningVkt(repositioningVkt);
+        model.setReservationCount(reservationCount);
+        GraphUtils.RIDE_HAIL_REVENUE_MAP.put(event.getIteration(), model);
+
+
     }
-    private List<String> getLegendItemList(String graphName,int dataSetRowCount,int bucketSize){
-        List<String> legendItemList = new ArrayList<>();
-        for (int i = 0; i < dataSetRowCount; i++) {
-            legendItemList.add(getLegendText(graphName, i,bucketSize));
+
+    private double[][] buildDeadHeadingDataSetTnc0() {
+        List<Integer> hours = GraphsStatsAgentSimEventsListener.getSortedIntegerList(deadHeadingsTnc0Map.keySet());
+        int maxHour = 0;
+        if (hours.size() > 0) {
```

Comment:
⚠️ Consider using `Collections.isEmpty()` instead of `Collections.size() > 0`. It increases the readability and also the complexity time can be reduced depending on the concrete class

## Suggestion

```Java
if (!hours.isEmpty()) {
```